platform/core/src/classes/MetadataProvider.js

```diff
@@ -445,15 +445,16 @@ class MetadataProvider {
     // check if the imageId starts with http:// or https:// using regex
     // Todo: handle non http imageIds
     let imageURI;
-    const urlRegex = /^(http|https):\/\//;
+    const urlRegex = /^(http|https|dicomfile):\/\//;
     if (urlRegex.test(imageId)) {
       imageURI = imageId;
     } else {
       imageURI = imageIdToURI(imageId);
     }
 
     const uids = this.imageURIToUIDs.get(imageURI);
-    const frameNumber = imageId.split(/\/frames\//)[1];
+    const frameNumber =
+      imageId.split(/\/frames\//)[1] || imageId.split('&frame=')[1];
```

Comment:
This doesn't work as you expect unless the frames/{...} or the &frame= is the last thing in the URL, but there is no guarantee that is so - particularly for &frame=, that could be anywhere in the URL.  You should write a URL that matches a sub-part, and then use that to split things up.